lib/cocoapods/sources_manager.rb

```diff
@@ -192,6 +214,44 @@ def search_index_path
       extend Executable
       executable :git
 
+      # Updates the stored search index if there are changes in spec repos while updating them.
+      # Update is performed incrementally. Only the changed pods' search data is re-generated and updated.
+      # @param  [Hash{Source => Array<String>}] changed_spec_paths
+      #                  A hash containing changed specification paths for each source.
+      def update_search_index_if_needed(changed_spec_paths)
+        search_index = nil
+        changed_spec_paths.each_pair do |source, spec_paths|
+          source_name = source.name
+          next unless spec_paths.length > 0
+          search_index = stored_search_index
+          updated_pods = source.pods_for_specification_paths(spec_paths)
+          new_index = aggregate.generate_search_index_for_changes_in_source(source, spec_paths)
+          next unless search_index && search_index[source_name]
+          # First traverse search_index and update existing words
+          # Removed traversed words from new_index after adding to search_index,
+          # so that only non existing words will remain in new_index after enumeration completes.
+          search_index[source_name].each_pair do |word, _|
```

Comment:
You could store the result of the `search_index[source_name]` access in `source_search_index` (example name) to clarify that it's always about the same piece of the search index.